app/common/build_artifact.py

```diff
@@ -140,8 +144,49 @@ def _update_timing_file(self, timing_file_path, new_timing_data):
         timing_data.update(new_timing_data)
         self._write_timing_data_to_file(timing_file_path, timing_data)
 
-    @staticmethod
-    def atom_artifact_directory(build_id, subjob_id, atom_id, result_root=None):
+    @classmethod
+    def get_console_output(
+            cls,
+            build_id: int,
+            subjob_id: int,
+            atom_id: int,
+            result_root: str,
+            max_lines: int=50,
+            offset_line: Optional[int]=None,
+    ) -> Optional[ConsoleOutputSegment]:
+        """
+        Return the console output if it exists in the specified result_root. Return None if it does not exist.
+        :param build_id: build id
+        :param subjob_id: subjob id
+        :param atom_id: atom id
+        :param result_root: the sys path to either the results or artifacts directory where results are stored.
+        :param max_lines: The maximum total number of lines to return. If this max_lines + offset_line lines do not
+            exist in the output file, just return what there is.
+        :param offset_line: The line number (0-indexed) to start reading content for. If none is specified, we will
+            return the console output starting from the end of the file.
+        :return: The console output if it exists in the specified result_root, None if it does not exist
+        """
+        console_output = None
+
+        artifact_dir = cls.atom_artifact_directory(build_id, subjob_id, atom_id, result_root=result_root)
+        output_file_path = os.path.join(artifact_dir, cls.OUTPUT_FILE)
+        if os.path.isfile(output_file_path):
+            # Read directly from output file if it exists (while build is in progress).
+            console_output = ConsoleOutput.from_plaintext(output_file_path)
+        else:
```

Comment:
Having this distinction between build artifacts that have been zipped, and build artifacts that haven't been zipped, makes me think that eventually we'll want to redefine, and split up, this "BuildArtifact" class into two different classes.

Definitely outside the scope of this PR.